test/evergreen/compatibility_test_for_mongodb_releases.sh

```diff
@@ -1,131 +1,134 @@
+        args+="in_memory=0 "                    # Interested in the on-disk format
+        args+="leak_memory=1 "                  # Faster runs
+        args+="logging_compression=snappy "     # We only built with snappy, force the choice
+        args+="rebalance=0 "                    # Faster runs
+        args+="rows=1000000 "
+        args+="salvage=0 "                      # Faster runs
+        args+="timer=4 "
+        args+="verify=0 "                       # Faster runs
+
+        for am in $2; do
+            dir="RUNDIR.$am"
+            echo "./t running $am access method..."
+            ./t -1q -h $dir "file_type=$am" $args
+        done
 }
 
-run()
+EXT="extensions=["
+EXT+="ext/compressors/snappy/.libs/libwiredtiger_snappy.so,"
+EXT+="ext/collators/reverse/.libs/libwiredtiger_reverse_collator.so, "
+EXT+="ext/encryptors/rotn/.libs/libwiredtiger_rotn.so, "
+EXT+="]"
+
+#############################################################
+# verify_backward:
+#       arg1: release #1
+#       arg2: release #2
+#       arg3: access methods list
+#############################################################
+verify_backward()
 {
-	# Build test files from each release.
-	(build_rel 3.4) || return 1
-	(build_rel 3.6) || return 1
-	(build_rel 4.0) || return 1
-	(build_rel 4.2) || return 1
-	(build_rel develop) || return 1
-
-	# Verify forward/backward compatibility.
-	(verify 3.4 3.6) || return 1
-	(verify 3.6 4.0) || return 1
-	(verify 4.0 4.2) || return 1
-	(verify 4.2 develop) || return 1
-
-	return 0
+        echo "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-="
+        echo "Release \"$1\" verifying \"$2\""
+        echo "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-="
+        
+        cd "$1"
+        for am in $3; do
+            dir="$2/test/format/RUNDIR.$am"
+            echo "$1/wt verifying $2 access method $am..."
+
+            WIREDTIGER_CONFIG="$EXT" ./wt -h "../$dir" verify table:wt
+        done
 }
 
 # Create a directory in which to do the work.
 top="test-compatibility-run"
-rm -rf $top && mkdir $top && cd $top || {
-	echo "$0: unable to create $top working directory"
-	exit 1
-}
-
-run
-exit $?
+rm -rf "$top" && mkdir "$top"
+cd "$top"
+
+# Build a release with a known name, then use it to figure out tagged release names.
+(build_release "develop")
+
+# Figure out the names of the releases we'll compare.
+cd develop
+rel34="$(get_tagged_release "3.4")"
+rel36="$(get_tagged_release "3.6")"
+rel40="$(get_tagged_release "4.0")"
+#rel42="$(get_tagged_release "4.2")"    XXX cannot handle 4.4 formats yet.
+rel42="mongodb-4.2"
+cd ..
+
+# Build the rest of the releases.
+(build_release "$rel34")
+(build_release "$rel36")
+(build_release "$rel40")
+(build_release "$rel42")
+
+# Run format in each release for supported access methods.
+(run_format "$rel34" "fix row var")
+(run_format "$rel36" "fix row var")
+(run_format "$rel40" "fix row var")
+(run_format "$rel42" "fix row var")
+(run_format "develop" "row")
```

Comment:
Is running format here used as a sanity test for each release version? They are not directly related to compatibility testing.

Reply:
We run `format` to generate the objects we then verify. For example, we run format in 4.2 to generate some objects that we then use the `develop` build to verify, to get confidence `develop` can read objects created by a 4.2 release.

Reply:
Okay, so running format is a preparation step for the later verify. Thanks.